iceoryx_hoofs/source/posix_wrapper/unix_domain_socket.cpp

```diff
@@ -178,7 +178,14 @@ expected<IpcChannelError> UnixDomainSocket::closeFileDescriptor() noexcept
         {
             if (IpcChannelSide::SERVER == m_channelSide)
             {
-                unlink(&(m_sockAddr.sun_path[0]));
+                auto unlinkCall = posixCall(unlink)(&(m_sockAddr.sun_path[0]))
```

Comment:
@zmostafa Could you update the year in the copyright header at the very top?

Reply:
@mossmaurice for testing sad path, the `closeFileDescriptor` method is private and called via `destroy` or `initalizeSocket`

- `destroy()` is also private and called by the destructor
-  `initalizeSocket` calls `closeFileDescribtor` and masks the error coming from it.

I think the proper way is to test `closeFileDescriptor` by calling it directly, but this might require some code refactoring in the `unitx_domain_socket` by adding a friend test function or so. 

What do you think ?  did I get something wrong ?

Reply:
@zmostafa Ok, I see. Well, considering that we will move the `UnixDomainSocket` class anyway to `iceoryx_dust`, where we have not quite as strict quality requirements, no need to create those tests here. Also @elfenpiff will work on mocking the POSIX API as part of a testing platform for `v4.0`.